src/argus_htmx/themes/utils.py

```diff
@@ -1,21 +1,50 @@
+from importlib.resources import files
+from pathlib import Path
+
 from django.conf import settings
-from argus_htmx import settings as argus_htmx_settings
+from django.core.exceptions import ImproperlyConfigured
 
+from argus_htmx import settings as default_htmx_settings
 
-def get_themes():
-    return getattr(settings, "DAISYUI_THEMES", argus_htmx_settings.DAISYUI_THEMES)
 
+__all__ = ["get_theme_names"]
 
-def get_theme_names():
-    themes = get_themes()
+
+def get_themes_from_setting():
+    themes_setting = getattr(settings, "DAISYUI_THEMES", default_htmx_settings.DAISYUI_THEMES)
     theme_names = []
-    for theme in themes:
+    for theme in themes_setting:
         if isinstance(theme, str):
             theme_names.append(theme)
         elif isinstance(theme, dict):
             theme_names.extend(theme.keys())
     return theme_names
 
 
+def get_themes_from_css():
+    static_url = Path(settings.STATIC_URL).relative_to("/")
+    stylesheet_path = static_url / default_htmx_settings.STYLESHEET_PATH
+    styles_css = files("argus_htmx").joinpath(stylesheet_path).read_text()
+    styles_css_lines = styles_css.split("{")
+    theme_names = []
+    for line in styles_css_lines:
+        if "data-theme=" not in line:
+            continue
+        _, after = line.split("=", 1)
+        theme_name, _ = after.split("]", 1)
+        theme_names.append(theme_name.strip())
```

Comment:
wouldn't it be cleaner and less fragile to use an `re.findall` search here?

Reply:
Easier to read for others, probably yes. I like to avoid regexes when I can.

Reply:
which is a good sentiment, but sometimes they make life easier and can actually make sense ;)